lib/ui/painting/image_decoder_impeller.cc

```diff
@@ -370,32 +373,44 @@ ImageDecoderImpeller::UploadTextureToShared(
   texture->SetLabel(impeller::SPrintF("ui.Image(%p)", texture.get()).c_str());
 
   if (texture_descriptor.mip_count > 1u && create_mips) {
-    auto command_buffer = context->CreateCommandBuffer();
-    if (!command_buffer) {
-      std::string decode_error(
-          "Could not create command buffer for mipmap generation.");
-      FML_DLOG(ERROR) << decode_error;
-      return std::make_pair(nullptr, decode_error);
-    }
-    command_buffer->SetLabel("Mipmap Command Buffer");
-
-    auto blit_pass = command_buffer->CreateBlitPass();
-    if (!blit_pass) {
-      std::string decode_error(
-          "Could not create blit pass for mipmap generation.");
-      FML_DLOG(ERROR) << decode_error;
-      return std::make_pair(nullptr, decode_error);
-    }
-    blit_pass->SetLabel("Mipmap Blit Pass");
-    blit_pass->GenerateMipmap(texture);
-
-    blit_pass->EncodeCommands(context->GetResourceAllocator());
-    if (!command_buffer->SubmitCommands()) {
-      std::string decode_error("Failed to submit blit pass command buffer.");
-      FML_DLOG(ERROR) << decode_error;
-      return std::make_pair(nullptr, decode_error);
+    std::optional<std::string> decode_error;
+#if FML_OS_IOS
+    gpu_disabled_switch->Execute(fml::SyncSwitch::Handlers().SetIfFalse(
```

Comment:
Should the true branch be setting `decode_error`?

Reply:
We ripped out opengl for iOS right?

Reply:
Correct